tests/validation/tests/v3_api/test_istio.py

```diff
@@ -0,0 +1,273 @@
+        assert result == '<title>Simple Bookstore App</title>\n'
+    return result
+
+
+def wait_for_wl_by_id_to_active(client, workload_uuid,
+                                timeout=DEFAULT_TIMEOUT):
+    start = time.time()
+    workloads = client.list_workload(uuid=workload_uuid).data
+    wl = workloads[0]
+    while wl['state'] != "active":
+        if time.time() - start > timeout:
+            raise AssertionError(
+                "Timed out waiting for state to get to active")
+        time.sleep(.5)
+        workloads = client.list_workload(uuid=workload_uuid).data
+        assert len(workloads) == 1
+        wl = workloads[0]
+    return wl
+
+
+def create_and_verify_istio_app(p_client, ns, project, name):
+    answers = {
+        "enableCRDs": "true",
+        "gateways.istio-ingressgateway.type": "NodePort",
+        "gateways.istio-ingressgateway.ports[0].nodePort":
+            ISTIO_INGRESSGATEWAY_NODEPORT,
+        "gateways.istio-ingressgateway.ports[0].port": 80,
+        "gateways.istio-ingressgateway.ports[0].targetPort": 80,
+        "gateways.istio-ingressgateway.ports[0].name": "http2",
+    }
+    external_id = "catalog://?catalog=" + ISTIO_CATALOG + \
+                  "&template=" + ISTIO_TEMPLATE + \
+                  "&version=" + ISTIO_VERSION + \
+                  "&namespace=cattle-global-data"
+    print("creating istio catalog app")
+    app = p_client.create_app(
+        name=name,
+        externalId=external_id,
+        targetNamespace=ns.name,
+        projectId=project.id,
+        answers=answers
+    )
+    print("verify istio app installed condition")
+    wait_for_condition(p_client, app, check_condition('Installed', 'True'), 45)
+    print("verify istio app deployment condition")
+    wait_for_condition(p_client, app, check_condition('Deployed', 'True'), 600)
+    return app
+
+
+def verify_admission_webhook():
+    has_admission_webhook = execute_kubectl_cmd(
+        'api-versions | grep admissionregistration', False)
+    if len(has_admission_webhook) == 0:
+        raise AssertionError(
+            "MutatingAdmissionWebhook and ValidatingAdmissionWebhook plugins "
+            "are not listed in the kube-apiserver --enable-admission-plugins")
+
+
+def create_health_check(p_client, ns):
+    workload = create_workload(p_client, ns)
+    workload = p_client.reload(workload)
+
+    # wait 15 seconds to run livenessProbe check
+    time.sleep(15)
+    results = validate_workload(p_client, workload, "deployment", ns.name)
+    assert results[0]['status']['containerStatuses'][0]['restartCount'] == 0
+    p_client.delete(workload)
+
+
+def add_istio_label_to_ns(c_client, ns):
+    labels = {
+        "istio-injection": "enabled"
+    }
+    ns = c_client.update_by_id_namespace(ns.id, labels=labels)
+    return ns
+
+
+def verify_istio_crds_count():
+    get_crd_counts = execute_kubectl_cmd(
+        'get crds | grep "istio.io" | wc -l', False)
+    assert get_crd_counts == '53\n'
```

Comment:
This value should be stored in a variable, and the comparison values should be stripped of whitespace beforehand. When running locally I get a failure due to whitespace mismatch in the string.

Why is this 53 in particular? Is this likely to change in the future?

Reply:
1. updated with set a variable and removed the trailing newline.
2. yes, it is committed to being 53 in `istio` v1.1.x versions, if we bump our istio chart version and the latest istio app update its CRD's [total count](https://istio.io/docs/setup/kubernetes/install/helm/), then we should update accordingly.